go/libraries/doltcore/sqle/writer/prolly_index_writer_keyless.go

```diff
@@ -160,7 +160,8 @@ func (k prollyKeylessWriter) uniqueKeyError(ctx context.Context, keyStr string,
 		vd := k.valBld.Desc
 		for from := range k.valMap {
 			to := k.valMap.MapOrdinal(from)
-			if existing[to], err = index.GetField(ctx, vd, from, value, k.mut.NodeStore()); err != nil {
+			// offset from index for keyless rows, as first field is the count
+			if existing[to], err = index.GetField(ctx, vd, from+1, value, k.mut.NodeStore()); err != nil {
```

Comment:
it might be better for valMap to just have the right mappings, embed the cardinality offset so that other methods don't have to add 1 everywhere

Reply:
That was my initial approach, but I don't think the `valMap` or the tuple descriptors can tell if the first field is encoding the count or a column of type uint64. Maybe it's worth modifying them? or changing `MapOrdinal` to take a keyless bool or something like that?

Reply:
something like this?
```go
func ordinalMappingsFromSchema(from sql.Schema, to schema.Schema) (km, vm val.OrdinalMapping) {
	keyless := schema.IsKeyless(to)
	km = makeOrdinalMapping(from, to.GetPKCols(), keyless)
	vm = makeOrdinalMapping(from, to.GetNonPKCols(), keyless)
	return
}
```